### PR TITLE
[WIP] chore: migrate to commander

### PR DESCRIPTION
**Pull request by rishabh3112**
Refers #717 

**What kind of change does this PR introduce?**
Migration to new arguments parser

**Did you add tests for your changes?**
No, needs verification

**If relevant, did you update the documentation?**
No, ideally shouldn't be needed

**Summary**
Use commander as decided at https://github.com/webpack/webpack-cli/issues/717#issuecomment-600083131

**Does this PR introduce a breaking change?**
Don't mean to, attempting as seamless transition as possible

**Other information**
Tasks needed to be done.
- [x] ~Handle custom types~ https://github.com/webpack/webpack-cli/blob/6eecb26ae8e1b2e032c925c21b2cca5e7f37a510/packages/webpack-cli/lib/utils/cli-flags.js#L284 ( flag values should be sanitised where their functionality is implemented )
- [ ] Migrate serve package to use `commander.js`
- [x] Add support for `webpack serve help`
- [ ] Verify tests
- [ ] Remove `command-line-args` completely
- [x] ~Process global flag correctly~ ( cli-team decided to drop support for it.)

### CODE REVIEW

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -91,6 +91,15 @@ class WebpackCLI extends GroupHelper {
     getCoreFlags() {
         return core;
     }
+
+    /**
+     * Convert camelcase to kebabcase
+     * @param {string} string
+     */
+    _toKebabCase(string) {
```
Comment (ryanclark) [inline packages/webpack-cli/lib/webpack-cli.js]: I think any helper function should live in a utils file, instead of being a part of the class

**Comment by anshumanv** [comment]
How's the progress on this? @rishabh3112

**Comment by evilebottnawi** [comment]
@anshumanv @rishabh3112 need help, I want to look at that on the next week, but feel free to go ahead

**Comment by rishabh3112** [comment]
I couldn't resolve the 3 tests which are failing. Need help in that.

**Comment by snitin315** [comment]
> I couldn't resolve the 3 tests which are failing. Need help in that.

I guess #1455 should fix `node-test` for you.

node

**Comment by anshumanv** [comment]
Closing in favour of https://github.com/webpack/webpack-cli/pull/1481

**Pull request closed without merging**